### PR TITLE
Adding VariantCall and related entities/properties

### PR DESCRIPTION
**Pull request by kreinold**
[DM-421](https://broadworkbench.atlassian.net/browse/DM-421 )

### CODE REVIEW

File: src/terra-core/TerraCoreDataModel.ttl [hunk 1]
```diff
@@ -2103,6 +2093,15 @@ TerraCore:hasGRCName
...
 .
+TerraCore:hasGeneContext
+  a owl:DatatypeProperty ;
+  rdfs:domain TerraCore:SequenceLocation ;
+  rdfs:label "has gene context" ;
+  rdfs:range xsd:anyURI ;
+  rdfs:range xsd:string ;
+  skos:definition "A specific gene context that applies to this variant." ;
+  TerraCore:hasColumnLabel "gene_context" ;
+.
```
Comment (rsc3) [inline src/terra-core/TerraCoreDataModel.ttl:2104]: Definition is recursive.

File: src/terra-core/TerraCoreDataModel.ttl [hunk 2]
```diff
@@ -2745,6 +2764,14 @@ TerraCore:hasStrain
   skos:definition "Text string to represent the strain." ;
   skos:prefLabel "hasStrain" ;
 .
+TerraCore:hasStructuralVariationType
+  a owl:DatatypeProperty ;
+  rdfs:domain TerraCore:VariantCall ;
+  rdfs:label "hasVariantReference" ;
+  rdfs:range xsd:anyURI ;
```
Comment (rsc3) [inline src/terra-core/TerraCoreDataModel.ttl:2771]: rdfs:range [
      owl:unionOf (
          xsd:anyURI
          xsd:string
        ) ;
    ] ;

File: src/terra-core/TerraCoreDataModel.ttl [hunk 3]
```diff
@@ -2972,6 +2991,8 @@ TerraCore:usesReferenceAssembly
   rdfs:domain TerraCore:AlignmentFile ;
```
Comment (rsc3) [inline src/terra-core/TerraCoreDataModel.ttl:2991]:
```
  rdfs:range [
      owl:unionOf (
rdfs:domain TerraCore:AlignmentFile ;
  rdfs:domain TerraCore:SequenceFile ;
  rdfs:domain TerraCore:SequencingActivity ;
  rdfs:domain TerraCore:VariantCall ;
  rdfs:domain TerraCore:VariantCallingActivity ;
        ) ;
    ] ;  
```

File: src/terra-core/TerraCoreDataModel.ttl [hunk 4]
```diff
@@ -2972,6 +2991,8 @@ TerraCore:usesReferenceAssembly
...
   rdfs:domain TerraCore:SequenceFile ;
   rdfs:domain TerraCore:SequencingActivity ;
+  rdfs:domain TerraCore:VariantCall ;
+  rdfs:domain TerraCore:VariantCallingActivity ;
   rdfs:label "usesReferenceAssembly" ;
   rdfs:range xsd:anyURI ;
```
Comment (rsc3) [inline src/terra-core/TerraCoreDataModel.ttl:2997]: `  rdfs:range [
      owl:unionOf (
          xsd:anyURI
          xsd:string
        ) ;
    ] ;`

**Review by rsc3 — CHANGES_REQUESTED** [review]

File: src/terra-core/TerraCoreDataModel.ttl [hunk 5]
```diff
@@ -1338,17 +1328,17 @@ TerraCore:SequenceLocation
...
       a owl:Restriction ;
       owl:cardinality "1"^^xsd:nonNegativeInteger ;
-      owl:onProperty TerraCore:hasLocation ;
+      owl:onProperty TerraCore:hasStartPosition ;
     ] ;
   rdfs:subClassOf [
       a owl:Restriction ;
       owl:cardinality "1"^^xsd:nonNegativeInteger ;
-      owl:onProperty TerraCore:hasStartPosition ;
+      owl:onProperty TerraCore:hasStopPosition ;
```
[the remark this answers is not in the document]
Comment (kreinold) [reply src/terra-core/TerraCoreDataModel.ttl:1336]: We'll revisit this in a future sprint.

File: src/terra-core/TerraCoreDataModel.ttl [hunk 6]
```diff
@@ -2103,6 +2093,15 @@ TerraCore:hasGRCName
   rdfs:range xsd:string ;
   skos:prefLabel "hasGRCName" ;
 .
+TerraCore:hasGeneContext
+  a owl:DatatypeProperty ;
+  rdfs:domain TerraCore:SequenceLocation ;
+  rdfs:label "has gene context" ;
+  rdfs:range xsd:anyURI ;
```
[the remark this answers is not in the document]
Comment (kreinold) [reply src/terra-core/TerraCoreDataModel.ttl:2100]: Updated in next checkin

File: src/terra-core/TerraCoreDataModel.ttl [hunk 7]
```diff
@@ -1696,8 +1689,12 @@ TerraCore:hasAnatomicalRegion
   rdfs:domain TerraCore:BioSample ;
```
Comment (rsc3) [inline src/terra-core/TerraCoreDataModel.ttl:1689]: `  rdfs:domain [
    owl:unionOf (
TerraCore:BioSample
TerraCore:SingleCell
    );
  ];`

File: src/terra-core/TerraCoreDataModel.ttl [hunk 8]
```diff
@@ -1817,15 +1826,19 @@ TerraCore:hasChild
   rdfs:domain TerraCore:FamilyMember ;
```
Comment (rsc3) [inline src/terra-core/TerraCoreDataModel.ttl:1826]: rdfs:domain [ owl:unionOf ( TerraCore:FamilyMember TerraCore:HumanDonor ); ];

File: src/terra-core/TerraCoreDataModel.ttl [hunk 9]
```diff
@@ -2239,8 +2282,12 @@ TerraCore:hasImmunogen
   rdfs:domain TerraCore:AssayActivity ;
```
Comment (rsc3) [inline src/terra-core/TerraCoreDataModel.ttl:2282]: rdfs:domain [ owl:unionOf (TerraCore:AssayActivity TerraCore:SampleTreatmentActivity ); ];

File: src/terra-core/TerraCoreDataModel.ttl [hunk 10]
```diff
@@ -2412,8 +2471,12 @@ TerraCore:hasOrganismType
   rdfs:domain TerraCore:Donor ;
```
Comment (rsc3) [inline src/terra-core/TerraCoreDataModel.ttl:2471]: rdfs:domain [ owl:unionOf (TerraCore:Donor TerraCore:FamilyMember ); ];

File: src/terra-core/TerraCoreDataModel.ttl [hunk 11]
```diff
@@ -2593,14 +2668,27 @@ TerraCore:hasReadLength
...
+  rdfs:domain TerraCore:VariantCall ;
+  rdfs:label "hasReferenceAllele" ;
+  rdfs:range xsd:string ;
+  skos:definition "The state of a molecule at a identified location." ;
+  skos:prefLabel "has reference allele" ;
+.
 TerraCore:hasReportedEthnicity
   a owl:DatatypeProperty ;
   rdfs:comment "Recommend using HANCESTRO ancestry category subclasses but a text string is also acceptable." ;
   rdfs:domain TerraCore:FamilyMember ;
```
Comment (rsc3) [inline src/terra-core/TerraCoreDataModel.ttl:2683]: rdfs:domain [ owl:unionOf (TerraCore:Donor TerraCore:FamilyMember ); ];

File: src/terra-core/TerraCoreDataModel.ttl [hunk 12]
```diff
@@ -2753,8 +2868,12 @@ TerraCore:hasTarget
   rdfs:domain TerraCore:AssayActivity ;
```
Comment (rsc3) [inline src/terra-core/TerraCoreDataModel.ttl:2868]: rdfs:domain [ owl:unionOf (TerraCore:AssayActivity TerraCore:SampleTreatmentActivity  ); ];

File: src/terra-core/TerraCoreDataModel.ttl [hunk 13]
```diff
@@ -2972,11 +3087,16 @@ TerraCore:usesReferenceAssembly
   rdfs:domain TerraCore:AlignmentFile ;
   rdfs:domain TerraCore:SequenceFile ;
```
Comment (rsc3) [inline src/terra-core/TerraCoreDataModel.ttl:3088]: rdfs:domain [ owl:unionOf (TerraCore:SequencingActivity TerraCore:VariantCall TerraCore:VariantCallingActivity TerraCore:AlignmentFile); ];

**Review by rsc3 — CHANGES_REQUESTED** [review]

File: src/terra-core/TerraCoreDataModel.ttl [hunk 14]
```diff
@@ -11,6 +11,7 @@
 @prefix : <https://datamodel.terra.bio/TerraCore#> .
 @prefix NCBITaxon_Organisms: <https://datamodel.terra.bio/NCBITaxon_Organisms_subset#> .
 @prefix OBIAssay: <https://datamodel.terra.bio/OBIAssay_subset#> .
+@prefix OBI_assaySubset: <https://datamodel.terra.bio/OBI_assaySubset#> .
```
[the remark this answers is not in the document]
Comment (kreinold) [reply src/terra-core/TerraCoreDataModel.ttl]: Fixed

File: src/terra-core/TerraCoreDataModel.ttl (hunk 4 above)
Comment (kreinold) [reply src/terra-core/TerraCoreDataModel.ttl]: next checkin fixes this

File: src/terra-core/TerraCoreDataModel.ttl (hunk 7 above)
Comment (kreinold) [reply src/terra-core/TerraCoreDataModel.ttl:1689]: next checkin resolves this

**Pull request merged**